src/audio/copier/copier_host.c

```diff
@@ -107,15 +107,14 @@ void copier_host_dma_cb(struct comp_dev *dev, size_t bytes)
 	if (cd->hd->copy_type == COMP_COPY_ONE_SHOT)
 		host_common_one_shot(cd->hd, bytes);
 
-	/* apply attenuation since copier copy missed this with host device remove */
-	if (cd->attenuation) {
-		if (dev->direction == SOF_IPC_STREAM_PLAYBACK)
-			sink = buffer_acquire(cd->hd->local_buffer);
-		else
-			sink = buffer_acquire(cd->hd->dma_buffer);
-
-		frames = bytes / get_sample_bytes(audio_stream_get_frm_fmt(&sink->stream));
-		frames = frames / audio_stream_get_channels(&sink->stream);
+	/* Apply attenuation since copier copy missed this with host device
+	 * remove. Attenuation has to be applied in HOST Copier only with
+	 * playback scenario.
+	 */
+	if (cd->attenuation && dev->direction == SOF_IPC_STREAM_PLAYBACK) {
+		source = buffer_acquire(cd->hd->dma_buffer);
+		sink = buffer_acquire(cd->hd->local_buffer);
+		frames = bytes / audio_stream_frame_bytes(&source->stream);
```

Comment:
seems there is no issue with previous implementation, at this moment, copy already done, all the data is already in sink buffer, attenuation only happens on sink buffer, so use sink->stream to calculate frame is more suitable?

Reply:
@btian1 We're getting number of bytes from DMA here. So if e.g. host format is 24/24bit/48000kHz/2ch there will be PCM conversion to 24/32bit applied on host copier. Using `sink->stream`  frame bytes  will result in 36 frames                 (288 bytes/4 * 2ch) to be attenuated instead of 48. Below is an example of distorted signal:
 
![image](https://github.com/thesofproject/sof/assets/93581130/708bc130-8444-4667-bd91-a64faafd49d9)

Reply:
ok, I see, is 24/24bit possible? or do you see this case? in CI or QB? I wonder it is still 24/32bit format.

Reply:
Yes, it is possible. It is one of the main formats used by Windows driver.